### PR TITLE
Don't expose bit flag operations quite so much to the user.

### PR DESCRIPTION
**Pull request by ltratt**
Using 'bitflags' gives us the same internal representation but slightly more static type safety.

### CODE REVIEW

File: ykpack/src/types.rs [hunk 1]
```diff
@@ -343,14 +344,17 @@ impl Display for Local {
...
-    /// This function is annotated #[interp_step].
-    pub const INTERP_STEP: u8 = 1 << 1;
-    /// This function is yktrace::trace_debug.
-    pub const TRACE_DEBUG: u8 = 1 << 2;
+bitflags! {
+    /// Bits in the `flags` bitfield in `Body`.
+    #[derive(Serialize, Deserialize)]
+    pub struct BodyFlags: u8 {
+        /// This function is annotated #[do_not_trace].
+        const DO_NOT_TRACE = 0b00000001;
```
Comment (vext01) [inline ykpack/src/types.rs:352]: can we use the `1<<X` idiom? It's easier to read and less error prone.

**Comment by vext01** [comment]
Just an easy comment. Looks good.

File: ykpack/src/types.rs (hunk 1 above)
Comment (ltratt) [reply ykpack/src/types.rs:352]: I used to use `1<<X` but I've made too many mistakes with it to be fond of it any more. To be honest, I didn't think deeply about this:  I think all of the examples you'll find using bitflags use the `0b` notation, which suggests that's what people have gravitated towards.

**Comment by ltratt** [comment]
@vext01 Does this PR need separate `bors`ing?

**Pull request merged**